File: fosso_backend/src/main/java/com/fosso/backend/fosso_backend/order/model/Order.java
package com.fosso.backend.fosso_backend.order.model;

import com.fosso.backend.fosso_backend.common.enums.OrderStatus;
import com.fosso.backend.fosso_backend.common.enums.PaymentMethod;
import com.fosso.backend.fosso_backend.user.model.Address;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Data
@Document(collection = "orders")
public class Order {
    @Id
    private String orderId;
    @Indexed
    private String orderTrackingNumber;
    private String customerId;
    private BigDecimal productsCost;
    private BigDecimal subtotal = BigDecimal.ZERO;
    private BigDecimal shippingCost = BigDecimal.ZERO;
    private BigDecimal total = BigDecimal.ZERO;
    private BigDecimal tax = BigDecimal.ZERO;
    private PaymentMethod paymentMethod;
    private LocalDateTime orderDateTime;
    private int deliveryDays;
    private LocalDate deliveryDate;
    private Address shippingAddress;
    private OrderStatus status;
    private List<OrderDetail> orderDetails = new ArrayList<>();
    private LocalDateTime createdDateTime;
    private LocalDateTime updatedDateTime;
}

File: fosso_frontend/src/types/order.ts
import type { Address, UserBriefDTO, UserDetailedDTO } from "../types/user";
import type { OrderStatus, PaymentMethod } from "../types/enums"; 
import type { ImageDTO } from "./image";

export interface OrderBriefDTO {
  orderId: string; 
  orderTrackingNumber: string;
  customerId: string; 
  total: number;
  items: number;
  deliveryDays: number; 
  deliveryDate: string; 
  orderStatus: OrderStatus;
  orderDateTime: string; 
}

export interface OrderDetail {
  merchantId: string; 
  productId: string; 
  productName: string; 
  quantity: number; 
  color: string; 
  size: string; 
  price: number; 
  shippingCost: number;
  subtotal: number;  
  orderTrack?: OrderTrack;
  user?: UserBriefDTO
}

export interface OrderDetailedDTO {
  orderId: string; 
  orderTrackingNumber: string;
  customerId: string; 
  productsCost: number;
  subtotal: number;
  shippingCost: number;
  total: number; 
  deliveryDays: number;
  deliveryDate: string;
  status: OrderStatus; 
  shippingAddress: Address; 
  orderDetails: OrderDetail[];
  orderDateTime: string; 
}

export interface OrderStatusUpdateRequest {
  status: OrderStatus; 
  notes: string; 
}

export interface CheckoutRequest {
  addressId: string; 
  paymentMethod: PaymentMethod;
}

export interface OrderTrack {
  updatedTime: string; 
  status: OrderStatus; 
  notes: string; 
}

export interface OrderMerchantDTO {
  orderId: string;
  orderTrackingNumber: string;
  customerId: string;
  merchantId: string;
  productId: string;
  productName: string;
  productImage: ImageDTO;
  quantity: number;
  color: string;
  size: string;
  price: number;
  subtotal: number;
  shippingCost: number;
  orderTrack: OrderTrack;
  paymentMethod: PaymentMethod;
  orderDateTime: string;
  deliveryDays: number;
  deliveryDate: string;
  shippingAddress: Address;
}